src/stores/problem_sets.ts

```diff
@@ -268,16 +272,21 @@ export const useProblemSetStore = defineStore('problem_sets', {
 		/**
 		 * Deletes the given UserSet from the store and the database.
 		 */
-		async deleteUserSet(user_set: UserSet) {
+		async deleteUserSet(user_set: UserSet): Promise<void> {
 			const course_id = useSessionStore().course.course_id;
 			const response = await
 			api.delete(`courses/${course_id}/sets/${user_set.set_id}/users/${user_set.course_user_id ?? 0}`);
-			// TODO: check for errors
-			const deleted_user_set = parseDBUserSet(response.data as ParseableDBUserSet);
-			const index = this.db_user_sets.findIndex(s => s.user_set_id === deleted_user_set.user_set_id);
-			this.db_user_sets.splice(index, 1);
-			user_set.set(deleted_user_set.toObject());
-			return user_set;
+			if (response.status === 200) {
+				const index = this.db_user_sets.findIndex((set) => set.user_set_id === user_set.user_set_id);
+				if (index < 0) {
+					logger.error('[user store/deleteUserSet]: the user set was not found in the store');
```

Comment:
Shouldn't we be checking that the user set is present before making the API call? This is an error that 'should never happen', but it doesn't make sense to check afterwards.

Also, I worry about having removed the Exceptions on the DB side for attempting to delete objects that don't exist... that should be an error still, and the response wouldn't be 200 OK.

Reply:
Will switch the order of these.  About the exception, see my comment below.